source/kraken/worker.cpp

```diff
@@ -737,6 +737,38 @@ pbnavitia::Response Worker::graphical_isochrone(const pbnavitia::GraphicalIsochr
 
 }
 
+pbnavitia::Response Worker::car_co2_emission_on_crow_fly(const pbnavitia::CarCO2EmissionRequest& request) {
+    const auto data = data_manager.get_data();
+    init_worker_data(data);
+
+    auto get_geographical_coord = [&](const pbnavitia::LocationContext& location){
+        std::cout << location.place() << std::endl;
+        auto origin_type = data->get_type_of_id(location.place());
+        auto origin = type::EntryPoint(origin_type, location.place(), location.access_duration());
+
+        if (origin_type == type::Type_e::Address || origin_type == type::Type_e::Admin
+                || origin_type == type::Type_e::StopArea || origin_type == type::Type_e::StopPoint
+                || origin_type == type::Type_e::POI) {
+            auto coordinates = coord_of_entry_point(origin, data);
+            return navitia::type::GeographicalCoord{coordinates.lon(), coordinates.lat()};
+        }
+        return navitia::type::GeographicalCoord{};
+    };
+    auto origin = get_geographical_coord(request.origin());
+    auto destin = get_geographical_coord(request.destination());
+    auto distance = origin.distance_to(destin);
+    pbnavitia::Response r;
+    auto co2_emission = r.mutable_car_co2_emission();
+
+    if (data->pt_data->physical_modes_map["physical_mode:Car"]->co2_emission) {
```

Comment:
it should exist for data provided by fusio, sometime we don't use fusio, typically for opendata on country not using latin alphabet since fusio can't support these.
So I propose to deal with **the reality**.